### PR TITLE
Quarantine flakiest tests

### PR DESCRIPTION
**Pull request by fgimenez**
Signed-off-by: Federico Gimenez <fgimenez@redhat.com>

**What this PR does / why we need it**: apply the `[QUARANTINE]` label to the flakiest tests at the moment, these tests won't be executed on the required lanes and will be run on separated non-required lanes added in this PR https://github.com/kubevirt/project-infra/pull/1003 

The quarantined tests have been selecting taking into account the most flaky tests as reported by testgrid here:
https://testgrid.k8s.io/kubevirt-presubmits#pull-kubevirt-e2e-k8s-1.17&width=20&show-stale-tests=
https://testgrid.k8s.io/kubevirt-presubmits#pull-kubevirt-e2e-k8s-1.17-rook-ceph&width=20&show-stale-tests=
https://testgrid.k8s.io/kubevirt-presubmits#pull-kubevirt-e2e-k8s-1.18&width=20&show-stale-tests=
https://testgrid.k8s.io/kubevirt-presubmits#pull-kubevirt-e2e-k8s-1.19&width=20&show-stale-tests= 
https://testgrid.k8s.io/kubevirt-presubmits#pull-kubevirt-e2e-kind-1.17-sriov&width=20&show-stale-tests=

The main goal is to start unblocking the current bottleneck we have in the CI queue by implementing one of the proposed solutions for addressing flakiness. We don't have any policy in place for removing the quarantined tests from the test suite or bringing them back to the required lanes, we should agree in any case that they are fixed before that happens.

At the moment the quarantine lanes are not reporting status to the PRs to not cause unnecessary noise before this PR is merged. Once it lands we will enable status report.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/operator_test.go
Comment (davidvossel) [inline tests/operator_test.go:1002]: we shouldn't quarantine this one. this is a critical test that guarantees our compatibility.  we're going to have to figure otu what's going on here. Is there an issue about this test's flakiness yet?

Comment (davidvossel) [reply tests/operator_test.go:1002]: it looks like this is related to the timing of the live migration for workload updates. i'm looking at it. 

```
tests/operator_test.go:1002
Timed out after 320.000s.
All VMIs should update via live migration
Expected
    <*errors.errorString | 0xc001dfca50>: {
        s: "waiting for migration b391ddf8-a330-453a-a3a8-e4322ae8b3b4 to complete for vmi kubevirt-test-default1/testvmi-csx2n",
    }
to be nil
tests/operator_test.go:653
```

Comment (fgimenez) [reply tests/operator_test.go:1002]: > Is there an issue about this test's flakiness yet?

Indeed https://github.com/kubevirt/kubevirt/issues/3610

Comment (davidvossel) [reply tests/operator_test.go:1002]: i'm working on this here, https://github.com/kubevirt/kubevirt/pull/5159

**Comment by fgimenez** [comment]
All green, @davidvossel can we merge this one ASAP so that other PRs are unblocked? Getting green on your PR will also be easier once this one is in.

**Review by davidvossel — COMMENTED** [review]
/approve
/hold

@fgimenez I approved but put this on hold just because of the update test. can we remove the update test from quarantine ```[QUARANTINE][test_id:3145]from previous release to target tested release```. Maybe cherry-pick my pr #5159 into this one if it helps. 

My concern here is if we happen to somehow introduce code that truly fails the update test, it can be a nightmare to unwind. Updates break in subtle ways sometimes.

**Comment by davidvossel** [comment]
feel free to remove the hold whenever. I just wanted to make sure my comment was read before this gets merged.

**Comment by fgimenez** [comment]
> Maybe cherry-pick my pr #5159 into this one if it helps.

@davidvossel makes total sense, cherry-picked, let's see how it goes.

File: tests/operator_test.go
Comment (dhiller) [reply tests/operator_test.go:1024]: If David's changes have been cherry picked, can't we remove the QUARANTINE here?

**Comment by fabiand** [comment]
@fgimenez can we add an ownership tag to every testcase in order to know who is responsible to fix the test?

File: tests/operator_test.go
Comment (fgimenez) [reply tests/operator_test.go]: Good catch @dhiller  removed quarantine label.

**Comment by fgimenez** [comment]
> @fgimenez can we add an ownership tag to every testcase in order to know who is responsible to fix the test?

@fabiand done, PTAL

**Review by dhiller — APPROVED** [review]

**Comment by fgimenez** [comment]
The pending tests have already passed in this PR, only waiting for them bc of retests:
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5156/pull-kubevirt-e2e-k8s-1.19/1369144656364834816
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5156/pull-kubevirt-e2e-k8s-1.18/1368846442071855104
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5156/pull-kubevirt-e2e-windows2016/1369031286928183296

Can we override them to unblock other PRs that are failing bc of the tests quarantined here?

**Comment by davidvossel** [comment]
> Can we override them to unblock other PRs that are failing bc of the tests quarantined here?

+1 from me. There's little risk in this PR and it has the potential to help quite a bit. the faster we get this in, the better. I think it's a good idea to side step the normal process to escalate this

**Pull request merged**
